src/rp2_common/hardware_clocks/clocks.c

```diff
@@ -245,10 +245,20 @@ void clock_gpio_init_int_frac(uint gpio, uint src, uint32_t div_int, uint8_t div
         invalid_params_if(HARDWARE_CLOCKS, true);
     }
 
+#if !PICO_RP2040 // assert currently broken on RP2040, but we know that hardware has 16-bit integer part
+    static_assert(CLOCKS_CLK_GPOUT0_DIV_INT_MSB - CLOCKS_CLK_GPOUT0_DIV_INT_LSB == 15, "");
+#endif
+    invalid_params_if(HARDWARE_CLOCKS, div_int >> 16);
```

Comment:
IMHO it looks super-weird to just see `div_int >> 16` as the condition here. I guess `div_int > ((1 << 16) - 1)` might be a bit more readable? (assuming that I've got my bit-shifting maths correct!)

Or perhaps just `(div_int >> 16) > 0` is "more obvious" than `div_int >> 16`? (Or maybe my brain just isn't in "C programming mode" right now :joy: )

Reply:
I think I prefer it as it is